cpu/kinetis/periph/ics.c

```diff
@@ -0,0 +1,197 @@
+**
+**     o Redistributions in binary form must reproduce the above copyright notice, this
+**       list of conditions and the following disclaimer in the documentation and/or
+**       other materials provided with the distribution.
+**
+**     o Neither the name of Freescale Semiconductor, Inc. nor the names of its
+**       contributors may be used to endorse or promote products derived from this
+**       software without specific prior written permission.
+**
+**     THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+**     ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+**     WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+**     DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+**     ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+**     (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+**     LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+**     ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+**     (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+**     SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**
+**     http:                 www.freescale.com
+**     mail:                 support@freescale.com
+**
+**     Revisions:
+**     - rev. 1.0 (2013-07-30)
+**         Initial version.
+**     - rev. 1.1 (2013-10-29)
+**         Definition of BITBAND macros updated to support peripherals with 32-bit acces disabled.
+**     - rev. 1.2 (2014-01-10)
+**         CAN - Corrected address of TSIDR1 register.
+**         CAN - Corrected name of MSCAN_TDLR bit DLC to TDLC.
+**         FTM0 - Added access macro for EXTTRIG register.
+**         NVIC - Registers access macros improved.
+**         SCB - Unused bits removed, mask, shift macros improved.
+**         Defines of interrupt vectors aligned to RM.
+**     - rev. 1.3 (2014-06-18)
+**         The declaration of clock configurations has been moved to separate header file system_MKE02Z2.h
+**         Module access macro {module}_BASES replaced by {module}_BASE_PTRS.
+**         I2C - Renamed status register S to S1 to match RM naming.
+**         Renamed interrupts: INT_PMC to INT_LVD_LVW, INT_ADC to ADC0,INT_WDOG to INT_Watchdog.
+**     - rev. 1.4 (2015-07-24)
+**         Correction of backward compatibility.
+**
+** ###################################################################
+*/
+
+/*!
+ * @file SKEAZ1284
+ * @version 1.4
+ * @date 2015-07-24
+ * @brief Device specific configuration file for SKEAZ1284 (implementation file)
+ *
+ * Provides a system configuration function and a global variable that contains
+ * the system frequency. It configures the device and initializes the oscillator
+ * (PLL) that is part of the microcontroller device.
+ */
+
+#include <stdint.h>
+#include "cpu.h"
```

Comment:
This file is taken from the reference library (or similar), right? You just added this `#include`, we may need to clarify and document this documentation, so if someone wants to update this file with a future release from the ref-lib, they know what to adapt for RIOT.

Reply:
Yes, I've just taken this from vendor's examples. Probably we should rewrite it at some point, but I just can't see an immediate value of such work.
Can it be documented in-line in `ics.c` or is there some dedicated place for that?

Reply:
@smlng  Added some documentation to ics.c